<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>590</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>700</width>
    <height>590</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Cube Solver</string>
  </property>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayoutNetwork">
    <item row="0" column="1">
     <widget class="QLineEdit" name="lineEditPort">
      <property name="text">
       <string>40000</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="labelPort">
      <property name="text">
       <string>Port</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>10</y>
     <width>541</width>
     <height>81</height>
    </rect>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEditTimeOut">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>180</y>
     <width>51</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>5000</string>
   </property>
   <property name="readOnly">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLabel" name="labelTimeOut">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>180</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Time out[ms]</string>
   </property>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_3">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>180</y>
     <width>122</width>
     <height>91</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_0">
    <item row="0" column="0">
     <widget class="QPushButton" name="color_0">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="color_2">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="color_4">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="color_5">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="color_3">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="color_1">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="color_6">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="color_7">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="color_8">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_5">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>380</y>
     <width>122</width>
     <height>91</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_1">
    <item row="0" column="0">
     <widget class="QPushButton" name="color_9">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="color_12">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="color_10">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="color_13">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="color_14">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="color_11">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="color_15">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="color_17">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="color_16">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_4">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>280</y>
     <width>122</width>
     <height>91</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QPushButton" name="color_18">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="color_19">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="color_21">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="color_20">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="color_22">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="color_23">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="color_24">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="color_25">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="color_26">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_8">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>280</y>
     <width>122</width>
     <height>91</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="1">
     <widget class="QPushButton" name="color_28">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="color_29">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="color_30">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="color_27">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="color_31">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="color_32">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="color_33">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="color_34">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="color_35">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_6">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>280</y>
     <width>122</width>
     <height>91</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="1" column="1">
     <widget class="QPushButton" name="color_40">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="color_41">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="color_37">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="color_38">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="color_39">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="color_36">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="color_42">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="color_43">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="color_44">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_7">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>280</y>
     <width>122</width>
     <height>91</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <widget class="QPushButton" name="color_45">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="color_47">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="color_46">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="color_49">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="color_48">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="color_50">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="color_51">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="color_52">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="color_53">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="labelCubeState">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>100</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Cube State</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEditCubeState">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>100</y>
     <width>601</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="labelSolution">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>150</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Cube Solution</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEditSolution">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>150</y>
     <width>481</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="labelSyntax">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>120</y>
     <width>501</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Syntax : &quot;f:ccccccccc[space]&quot; x 6 ( f : face in order [U, D, L, R, F, B], c : color [W, O, R, B, G, Y] )</string>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>40</y>
     <width>131</width>
     <height>54</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QPushButton" name="pushButtonStart">
      <property name="text">
       <string>Server Start</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="pushButtonStop">
      <property name="text">
       <string>Server Stop</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QProgressBar" name="progressBar">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>300</y>
     <width>131</width>
     <height>171</height>
    </rect>
   </property>
   <property name="value">
    <number>0</number>
   </property>
   <property name="textVisible">
    <bool>false</bool>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QLabel" name="labelCurrentProgress">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>280</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Current Progress</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QCheckBox" name="checkBoxThrough">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>210</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Through Mode Solve</string>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButtonSolve">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>240</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Solve</string>
   </property>
  </widget>
  <widget class="QTextBrowser" name="textBrowserSolver">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>480</y>
     <width>631</width>
     <height>101</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButtonSolverClear">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>480</y>
     <width>41</width>
     <height>101</height>
    </rect>
   </property>
   <property name="text">
    <string>Clear</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="checkBoxBeacon">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>150</y>
     <width>61</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Beacon</string>
   </property>
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEditBeaconNum">
   <property name="geometry">
    <rect>
     <x>640</x>
     <y>150</y>
     <width>21</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="readOnly">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEditBeaconStr">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>150</y>
     <width>31</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="readOnly">
    <bool>false</bool>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
